src/wasm/literal.cpp

```diff
@@ -2684,36 +2685,30 @@ Literal Literal::relaxedFmsF64x2(const Literal& left,
 }
 
 Literal Literal::externalize() const {
-  assert(Type::isSubType(type, Type(HeapType::any, Nullable)) &&
+  auto heapType = type.getHeapType();
+  auto anyType = HeapTypes::any.getBasic(heapType.getShared());
+  assert(Type::isSubType(type, Type(anyType, Nullable)) &&
```

Comment:
This would require less work.
```suggestion
  assert(type.isRef() && type.getHeapType().getTop().isMaybeShared(HeapType::any) &&
```